cmake/catkin_download.cmake

```diff
@@ -0,0 +1,146 @@
+#
+# Download a file containing data from a URL.
+#
+# It is commonly used to download larger data files which should not be
+# stored in the repository.
+#
+# .. note:: It is not recommended to rely on downloaded data during
+#   a configure / make cycle since this prevents building the package
+#   when no network connectivity is available.
+#
+# .. note:: The target will be registered as a dependency
+#   of the "download_extra_data" target.
+#
+# :param target: the target name
+# :type target: string
+# :param url: the url to download
+# :type url: string
+#
+# :param DESTINATION: the directory where the file is downloaded to
+#   (default: ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION})
+# :type DESTINATION: string
+# :param FILENAME: the filename of the downloaded file
+#   (default: the basename of the url)
+# :type FILENAME: string
+# :param MD5: the expected md5 hash to compare against
+#   (default: empty, skipping the check)
+# :type MD5: string
+# :param EXCLUDE_FROM_ALL: exclude this download from the 'all' build target
+#   (default: true, excluding it from the all target)
+# :type EXCLUDE_FROM_ALL: bool
```

Comment:
In CMake `EXCLUDE_FROM_ALL` is usually used as an option. It might be confusing to use it as a boolean argument.